docs/en/creating-widgets/supplemental.md

```diff
@@ -0,0 +1,869 @@
+
+For large applications, state management can be one of the most challenging aspects to deal with, requiring developers to balance between data consistency, availability and fault tolerance. While a lot of this complexity remains outside the scope of the web application layer, Dojo provides further solutions that help ensure data consistency. The [Dojo Stores](../dojo-stores/supplemental.md) component provides a centralized state store with a consistent API for accessing and managing data from multiple locations within the application.
+
+## Basic: self-encapsulated widget state
+
+Widgets can maintain their own internal state as locally-scoped variables.
+
+Internal state data may directly affect the widget's render output, or may be passed as properties to any child widgets where they in turn directly affect the children's render output. Widgets may also allow their internal state to be changed, for example in response to a user interaction event.
+
+The following example illustrates these patterns:
+
+> src/widgets/MyEncapsulatedStateWidget.tsx
+
+**Function-based variant:**
+
+```tsx
+import { create, tsx } from '@dojo/framework/core/vdom';
+
+const factory = create();
+
+let myState: string = 'Hello from a stateful widget!';
+let counter: number = 0;
+
+export default factory(function MyEncapsulatedStateWidget() {
+	return (
+		<div>
+			Current widget state: {myState}
+			<br />
+			<button
+				onclick={() => {
+					myState = 'State change iteration #' + ++counter;
+				}}
+			>
+				Change State
+			</button>
+		</div>
+	);
+});
+```
+
+**Class-based variant:**
+
+```tsx
+import WidgetBase from '@dojo/framework/core/WidgetBase';
+import { tsx } from '@dojo/framework/core/vdom';
+
+export default class MyEncapsulatedStateWidget extends WidgetBase {
+	private myState: string = 'Hello from a stateful widget!';
+	private counter: number = 0;
+
+	protected render() {
+		return (
+			<div>
+				Current widget state: {this.myState}
+				<br />
+				<button
+					onclick={() => {
+						this.myState = 'State change iteration #' + ++this.counter;
+					}}
+				>
+					Change State
+				</button>
+			</div>
+		);
+	}
+}
+```
+
+Note that this example is not complete - clicking on the 'Change State' button in the running application will not have any effect on the widget's render output. This is because the state is fully encapsulated within `MyEncapsulatedStateWidget`, and Dojo [is not aware of any changes made to it](#widget-properties). Only the widget's initial render will be processed by the framework.
```

Comment:
@sbinge Ah okay ignore my comment about invalidating, if you are progressively enhancing this example - but you could move from `cache` to `icache` or just use `cache` and `invalidator` if the intent is to explicitly demo the `invalidator` middleware.